Library/Homebrew/language/node.rb

```diff
@@ -4,15 +4,27 @@ def self.npm_cache_config
       "cache=#{HOMEBREW_CACHE}/npm_cache\n"
     end
 
-    def self.pack_for_installation
+    def self.pack_for_installation(prepublish_requires_deps, pack_skip_prepublish)
+      # Some packages are requiring (dev)dependencies to be in place when
+      # running the prepublish script (which is run before pack). In this case
+      # we have to install all dependencies a first time already before
+      # executing npm pack (and a second time when doing the actual install).
+      safe_system "npm", "install" if prepublish_requires_deps
+
       # Homebrew assumes the buildpath/testpath will always be disposable
       # and from npm 5.0.0 the logic changed so that when a directory is
       # fed to `npm install` only symlinks are created linking back to that
       # directory, consequently breaking that assumption. We require a tarball
       # because npm install creates a "real" installation when fed a tarball.
-      output = Utils.popen_read("npm pack").chomp
-      raise "npm failed to pack #{Dir.pwd}" unless $CHILD_STATUS.exitstatus.zero?
-      output
+      pack_cmd = "npm pack"
+      # We don't want to run the prepublish script for some node modules in
+      # some formulae, so skip it by using --ignore-scripts.
+      pack_cmd += " --ignore-scripts" if pack_skip_prepublish
+      output = Utils.popen_read(pack_cmd)
```

Comment:
One thing I'd noticed while trying to get `now-cli` to work was that stderr for the pack command isn't getting printed, which made it a bit difficult to see what was going wrong. I wonder if there's a reasonable way to get stderr printed as well even though we need to capture stdout for the tarball name, or if we just need to continue to debug those situations by manually running `npm pack` within the `debrew` shell.